src/datasets.jl

```diff
@@ -694,6 +714,96 @@ function get_chunk(dset::Dataset)
     ret
 end
 
+struct ChunkInfo{N}
+    offset::NTuple{N,Int}
+    filter_mask::Cuint
+    addr::API.haddr_t
+    size::API.hsize_t
+end
+function Base.show(io::IO, ::MIME"text/plain", info::Vector{<:ChunkInfo})
+    print(io, typeof(info))
+    println(io, " with $(length(info)) elements:")
+    println(io, "Offset    \tFilter Mask                     \tAddress\tSize")
+    println(io, "----------\t--------------------------------\t-------\t----")
+    for ci in info
+        println(
+            io,
+            @sprintf("%10s", ci.offset),
+            "\t",
+            bitstring(ci.filter_mask),
+            "\t",
+            ci.addr,
+            "\t",
+            ci.size
+        )
+    end
+end
+
+"""
+    HDF5.get_chunk_info_all(dataset, [dxpl])
+
+Obtain information on all the chunks in a dataset. Returns a
+`Vector{ChunkInfo{N}}`.  The fields of `ChunkInfo{N}` are
+* offset - `NTuple{N, Int}` indicating the offset of the chunk in terms of elements, reversed to F-order
+* filter_mask - Cuint, 32-bit flags indicating whether filters have been applied to the cunk
+* addr - haddr_t, byte-offset of the chunk in the file
+* size - hsize_t, size of the chunk in bytes
+"""
+function get_chunk_info_all(dataset, dxpl=API.H5P_DEFAULT)
+    @static if hasmethod(API.h5d_chunk_iter, Tuple{API.hid_t})
+        return _get_chunk_info_all_by_iter(dataset, dxpl)
+    else
+        return _get_chunk_info_all_by_index(dataset, dxpl)
+    end
+end
+
+"""
+    _get_chunk_info_all_by_iter(dataset, [dxpl])
+
+Implementation of [`get_chunk_info_all`](@ref) via [`HDF5.API.h5d_chunk_iter`](@ref).
+
+We expect this will be faster, O(N), than using `h5d_get_chunk_info` since this allows us to iterate
+through the chunks once.
+"""
+@inline function _get_chunk_info_all_by_iter(dataset, dxpl=API.H5P_DEFAULT)
+    ds = dataspace(dataset)
+    N = ndims(ds)
+    info = ChunkInfo{N}[]
+    num_chunks = get_num_chunks(dataset)
+    sizehint!(info, num_chunks)
+    API.h5d_chunk_iter(dataset, dxpl) do offset, filter_mask, addr, size
+        _offset = reverse(unsafe_load(Ptr{NTuple{N,Int}}(offset)))
```

Comment:
Might be a good idea to use the correct types:
```suggestion
        _offset = reverse(unsafe_load(Ptr{NTuple{N,hsize_t}}(offset)))
```